mlflow/keras.py

```diff
@@ -385,33 +383,32 @@ class __MLflowKerasCallback(keras.callbacks.Callback):
         Records available logs after each epoch.
         Records model structural information as params when training begins
         """
+
         def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
-            try_mlflow_log(mlflow.log_param, 'num_layers', len(self.model.layers))
-            try_mlflow_log(mlflow.log_param, 'optimizer_name', type(self.model.optimizer).__name__)
+            try_mlflow_log(
+                mlflow.log_param, 'num_layers', len(
+                    self.model.layers))
+            try_mlflow_log(
+                mlflow.log_param, 'optimizer_name', type(
+                    self.model.optimizer).__name__)
             if hasattr(self.model.optimizer, 'lr'):
                 lr = self.model.optimizer.lr if \
-                    type(self.model.optimizer.lr) is float \
+                    isinstance(self.model.optimizer.lr, float) \
                     else keras.backend.eval(self.model.optimizer.lr)
                 try_mlflow_log(mlflow.log_param, 'learning_rate', lr)
             if hasattr(self.model.optimizer, 'epsilon'):
                 epsilon = self.model.optimizer.epsilon if \
-                    type(self.model.optimizer.epsilon) is float \
+                    isinstance(self.model.optimizer.epsilon, float) \
                     else keras.backend.eval(self.model.optimizer.epsilon)
                 try_mlflow_log(mlflow.log_param, 'epsilon', epsilon)
 
             sum_list = []
             self.model.summary(print_fn=sum_list.append)
             summary = '\n'.join(sum_list)
-            try_mlflow_log(mlflow.set_tag, 'model_summary', summary)
-
-            tempdir = tempfile.mkdtemp()
```

Comment:
The reason we wrote the code like this was because we don't want to write random files to the user's local directory. We temporarily create a `tempdir` to write `model_summary.txt` to, log the file as an artifact (which copies it over to the artifact uri), and then tear down the `tempdir` we created. Your implementation, while correct in end result, would leave `model_summary.txt` as a file in the local directory (which we want to avoid). I would recommend reverting this section.

Reply:
As mentioned in the previous comment this behavior has been changed and nothing is saved in the user's directory. It is only logged as an artifact if the model is larger than 5000 characters.